### PR TITLE
Enhance project configurations and improve error handling

### PR DESCRIPTION
**Pull request by GreenXeMotion**
- Added DebugType and NoWarn settings to project files for EpinelPS, ServerSelector, and ServerSelector.Desktop.
- Updated ServerCertificate initialization in Program.cs to read from file bytes for better reliability.
- Improved error handling in DoLimitBreak.cs and ExecuteEventGacha.cs by throwing exceptions for null references and unavailable characters, respectively.
- Refactored ColorConsoleLoggerProvider to simplify logger creation.

### CODE REVIEW

File: EpinelPS/EpinelPS.csproj [hunk 1]
```diff
@@ -9,6 +9,9 @@
 		<PublishSingleFile>true</PublishSingleFile>
 		<SelfContained>true</SelfContained>
 		<IncludeNativeLibrariesForSelfExtract>True</IncludeNativeLibrariesForSelfExtract>
+		<DebugType>none</DebugType>
```
Comment (MishaProductions) [inline EpinelPS/EpinelPS.csproj:12]: Can you please undo these changes?

**Comment by MishaProductions** [comment]
Everything else looks good, thanks!

File: EpinelPS/EpinelPS.csproj (hunk 1 above)
Comment (GreenXeMotion) [reply EpinelPS/EpinelPS.csproj]: Done.

**Comment by MishaProductions** [comment]
Thanks for the PR!
